httpclient/httpclient.go

```diff
@@ -183,28 +191,30 @@ func (c *HTTPConnectDialer) DialContext(ctx context.Context, network, address st
 		c.cacheH2Mu.Unlock()
 	}
 
-	rawConn, err := c.Dialer.DialContext(ctx, network, c.ProxyUrl.Host)
-	if err != nil {
-		return nil, err
-	}
-
+	var err error
+	var rawConn net.Conn
 	negotiatedProtocol := ""
 	switch c.ProxyUrl.Scheme {
 	case "http":
+		rawConn, err = c.Dialer.DialContext(ctx, network, c.ProxyUrl.Host)
+		if err != nil {
+			return nil, err
+		}
 	case "https":
+		var tlsConn TLSConn
 		if c.DialTLS != nil {
-			rawConn, negotiatedProtocol, err = c.DialTLS(rawConn)
+			tlsConn, err = c.DialTLS(network, c.ProxyUrl.Host)
 			if err != nil {
 				return nil, err
 			}
-			break
-		}
-		tlsConf := tls.Config{
-			NextProtos: []string{"h2", "http/1.1"},
-			ServerName: c.ProxyUrl.Hostname(),
-		}
+		} else {
+			tlsConf := tls.Config{
+				NextProtos: []string{"h2", "http/1.1"},
+				ServerName: c.ProxyUrl.Hostname(),
+			}
 
-		tlsConn := tls.Client(rawConn, &tlsConf)
+			tlsConn = tls.Client(rawConn, &tlsConf)
```

Comment:
I think this won't work, because `rawConn` will be `nil`.

Reply:
I am not sure if that's the case.
You might've missed the `rawConn = tlsConn` assignment at the end of `case https:`

Reply:
That happens _after_ this call, so `rawConn` will still be `nil` at this point.

Reply:
Oooooops. My bad.